prov/efa/src/efa_domain.c

```diff
@@ -232,26 +235,32 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 
 	*domain_fid = &domain->util_domain.domain_fid;
 
+	domain->cache = NULL;
 	/*
 	 * If FI_MR_LOCAL is set, we do not want to use the
 	 * MR cache
 	 */
-	if (info->domain_attr && info->domain_attr->mr_mode & FI_MR_LOCAL) {
-		efa_mr_cache_enable = 0;
-	}
+	enable_cache = efa_mr_cache_enable && info->domain_attr && info->domain_attr->mr_mode & ~FI_MR_LOCAL;
 
-	if (efa_mr_cache_enable && ofi_vrb_check_fork_enabled(*domain_fid)) {
+	if (enable_cache && ofi_vrb_check_fork_enabled(*domain_fid)) {
```

Comment:
Posting comment here as well.  The PR that added ofi_vrb_check_fork_enabled() results in a runtime error, I believe because libibverbs isn't linked.

https://github.com/ofiwg/libfabric/issues/6327

For now, the quick fix is to move that code back into the EFA provider (since verbs isn't using it at the moment).  We can figure out where to put shared verbs related code after v1.11.1 is released.